docs/developers/content-creation/creating-a-weapon.md

```diff
@@ -1,3 +1,120 @@
+
+* `weapon_ttt_<weapon name>\gamemodes\terrortown\entities\weapons\weapon_ttt_\<weapon name>.lua`
+
+Your .vmt and .vtf icon files (what gets shown in the buy menu) will need to be stored in this format:
+
+* `\weapon_ttt_<weapon name>\materials\vgui\ttt\<weapon name>_icon.vtf`
+* `\weapon_ttt_<weapon name>\materials\vgui\ttt\<weapon name>_icon.vmt`
+
+See the [Creating An Addon](https://docs.ttt2.neoxult.de/developers/basics/creating-an-addon/) page for more info on folder structure
+
+## Important 'Structural' features
+
+### If SERVER Block
+
+```lua
+if SERVER then
+   -- Shares the specified file with the client (if blank shares current file)
+   AddCSLuaFile()
+
+   -- Shares the buy menu icon with the client
+   resource.AddFile("materials/vgui/ttt/blue_template_icon.vmt")
+end
+```
+It's very important to create a globally unique name for this because the Gmod
+client will look for the given path locally to display it. So if there are two
+identical icon names the client may display the incorrect one
+
+### If CLIENT Block
+
+```lua
+if CLIENT then
+   -- Client viewpoint vals
+   SWEP.PrintName          = "Template Weapon"
+   SWEP.Slot               = 6 -- (See SWEP.Kind below)
+   SWEP.ViewModelFOV       = 70 -- How much of the first person viewmodel is seen
+   SWEP.ViewModelFlip      = true -- If true, the viewmodel is on the left
+   SWEP.DrawCrossHair      = false -- Does the client draw its own crosshair over the default ones
+
+   -- Client menu data
+   SWEP.Icon = "vgui/ttt/blue_template_icon.vtf" -- The filename of the icon shown in the buymenu
+   -- The information shown in the buy menu
+   SWEP.EquipMenuData      =
+   {
+      type = "Weapon",
+      desc = "This is a template weapon"
+   };
+end
+```
+
+
+
+## Important SWEP values
+
+```lua
+-- What weapon this is based upon (always use weapon_tttbase when creating a ttt/2 weapon)
+SWEP.Base                  = "weapon_tttbase"
+```
+The TTT2 gamemode has a file which describes the default values and methods. Any SWEP which specifies weapon_tttbase inherits the attributes and methods unless specified.
+
+```lua
+-- Primary fire vals
+SWEP.Primary.Delay         = 0.001 -- Higher is slower
+SWEP.Primary.Automatic     = true -- Is it automatic
+SWEP.Primary.Damage        = 1 -- How much damage per shot
+SWEP.Primary.Cone          = 0.5 -- Cone of accuracy
+SWEP.Primary.Ammo          = "smg1" -- Which ttt ammo entity is picked up by this swep
+SWEP.Primary.ClipSize      = 100 -- Maximum clip size
+SWEP.Primary.DefaultClip   = 100 -- Starting clip size
+SWEP.Primary.Sound         = Sound("Weapon_AK47.Single") -- Either a sound file from the default list or a custom one
+SWEP.Primary.NumShots      = 1 -- Number of shots fired at one time (make into shotgun if increase conde val)
+SWEP.Primary.Recoil        = 1.5 -- Recoil value (bigger = bigger)
+```
+Note that all of these attributes can also be applied to SWEP.Secondary which is alt click if you disable ironsights
```

Comment:
if you want to you could use the following syntax to create something like this: 
![image](https://user-images.githubusercontent.com/16392835/85954177-dd19f700-b975-11ea-8d3f-68a0a716d0f5.png)
```suggestion
!!! note
    All of these attributes can also be applied to SWEP.Secondary which is alt click if you disable ironsights
```

Reply:
I can't seem to replicate this effect using that syntax? Although I must admit I'm using the built in MD previewer in VSC

Reply:
This is a custom feature of MKDocs. This does not work in VSCode. Just take a look at our other pages and how we've done it there